### PR TITLE
Create super wide RHP modal to display expense reports when accessed via Reports page or report previews

### PR DESCRIPTION
**Pull request by WojtekBoman**
#### Super Wide RHP 

This PR introduces a third width of RHP called **Super Wide RHP**.

The maximum width of this RHP is `1260px`.

This type of RHP is available only for the following pages in the app:
- `SCREENS.EXPENSE_REPORT_RHP (e/:reportID)` - it's a new page added to the app. It's responsible for displaying expense reports in super wide / wide RHP in the Inbox tab
- `SCREENS.SEARCH.MONEY_REQUEST_REPORT (search/r/:reportID)`

Super wide RHP is displayed for these pages only if the open RHP has more than one visible transaction, otherwise these pages are opened in Wide RHP.

#### Overlays
There are two levels of nesting in RHP:
1. RightModalNavigator displays ModalStackNavigators (RHP flows).
2. ModalStackNavigators contain screens displayed in RHP.

After adding the third RHP width, it became necessary to add a third overlay displayed when we overlap Super Wide RHP -> Wide RHP -> RHP.

There are two levels of nesting in RHP:
1. Right Modal Navigator displays ModalStackNavigators (RHP flows).
2. ModalStackNavigators contain screens displayed in RHP.

This information is important for correctly adding a new overlay.

When a new overlay is added, it is needed to be adjusted in 2 files:
- `src/libs/Navigation/AppNavigator/Navigators/RightModalNavigator.tsx`

    Here, an overlay occupying the space outside the screen displayed in RHP must be added, when clicking on this overlay, the screen in RHP is closed.

- `src/libs/Navigation/AppNavigator/ModalStackNavigators/index.tsx`

    In case 2 or more RHPs of different widths are displayed, the RHP should also be added here.
    
    In `RightModalNavigator`, an overlay is shown from the left edge of the screen to the edge of the screen displayed as the modal. In `ModalStackNavigator`, the overlay should be displayed below the currently displayed modal. This way, when opening another modal, you can see that there's also an overlay underneath it.

More information about how overlays are displayed and their positioning can be found directly in the comments in the files mentioned above.

### Fixed Issues

$ https://github.com/Expensify/App/issues/71821
PROPOSAL:

### Tests

- [ ] Verify that no errors appear in the JS console

Initial conditions:
Create a report containing at least two expenses.

Test 1

1. Go to the Reports page in the Reports tab.
2. Click on the report you created.
3. Verify that it opens as a super wide RHP, no style bugs appear, the overlay works correctly, the backButton returns you back, and there are no other issues.

Test 2

1. Add two more reports: one empty, and one containing a single expense.
2. Go to the Reports page in the Reports tab.
3. Click on any report.
4. Switch between the reports you created using the arrows in the top-right corner.
5. Verify that:
- the report with one transaction displays as a wide RHP,
- the empty report also displays as a wide RHP (but without a receipt),
- when switching between reports, all styles update correctly,
- and the RHP sizes change depending on the number of transactions (0, 1, >1).

Test 3

1. Go to the Reports page in the Reports tab.
2. Open a report with more than one transaction.
3. Click on one of the expenses.
4. Verify that a wide RHP opens and overlays are displayed correctly.
5. Close the wide RHP by clicking on the overlay or using the backButton.
6. Verify that the wide RHP closed correctly, the overlay disappeared, and the UI returned to the super wide RHP.

Test 4

1. Go to the Inbox page and find a report with multiple expenses.
2. Click on it.
3. Verify that the SWRHP opens, nothing is broken, and everything functions correctly.

Test 5

1. Go to the Reports page in the Reports tab.
2. Open a report with more than one transaction.
3. Click on one of the expenses.
4. Click the More button in the wide RHP and choose an option that opens an RHP (e.g., Info).
5. Verify that the RHP opened correctly, overlays reflect the proper structure, and nothing is broken.

Test 6

1. Go to the Reports page in the Reports tab.
2. Open a report with more than one transaction.
3. Click on one of the expenses.
4. Click the link below the wide RHP title that leads back to the parent report.
5. Verify that the wide RHP correctly returns back to the parent super wide RHP.

### QA Steps

Same as tests

- [ ] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I verified there are no new alerts related to the `canBeMissing` param for `useOnyx`
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4510fc76bbf5df699a2575bfb49a276af90f3ed7/src/components/LocaleContextProvider.tsx#L80)
      - [x] If any non-english text was added/modified, I used [JaimeGPT](https://chatgpt.com/g/g-2dgOQl5VM-english-to-spanish-translator-aka-jaimegpt) to get English > Spanish translation. I then posted it in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4510fc76bbf5df699a2575bfb49a276af90f3ed7/src/components/LocaleContextProvider.tsx#L116-L123)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.ts or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG)`)
- [x] If new assets were added or existing ones were modified, I verified that:
    - [x] The assets are optimized and compressed (for SVG files, run `npm run compress-svg`)
    - [x] The assets load correctly across all supported platforms.
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] I added [unit tests](https://github.com/Expensify/App/blob/main/tests/README.md) for any new feature or bug fix in this PR to help automatically prevent regressions in this user flow.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

https://github.com/user-attachments/assets/51f7350e-822d-4e82-b3e4-23da70167c20

</details>

<details>
<summary>Android: mWeb Chrome</summary>

https://github.com/user-attachments/assets/84069892-ac46-4658-88d1-8707618a28e1

</details>

<details>
<summary>iOS: Native</summary>

https://github.com/user-attachments/assets/665b4aef-adae-439a-8f2d-3505715df527

https://github.com/user-attachments/assets/d596930f-06f2-40b8-9542-b466ece3349a

</details>

<details>
<summary>iOS: mWeb Safari</summary>

https://github.com/user-attachments/assets/e63bf433-a1c2-42f2-84da-585f2e45dbf8

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

https://github.com/user-attachments/assets/a32d42a0-50b6-4253-a424-339ebe4776ce

</details>

<details>
<summary>MacOS: Desktop</summary>

</details>

### CODE REVIEW

File: src/components/WideRHPContextProvider/index.tsx [hunk 1]
```diff
@@ -285,16 +377,43 @@ function WideRHPContextProvider({children}: React.PropsWithChildren) {
...
 
+    /**
+     * Effect that manages the secondary overlay animation and rendering state.
+     */
+    useEffect(() => {
+        if (shouldShowThirdOverlay) {
+            setShouldRenderThirdOverlay(true);
+            Animated.timing(thirdOverlayProgress, {
+                toValue: 1,
+                duration: 300,
```
Comment (ZhenjaHorbach) [inline src/components/WideRHPContextProvider/index.tsx]: I suppose it's better to use constants for that

File: src/libs/Navigation/AppNavigator/ModalStackNavigators/index.tsx [hunk 2]
```diff
@@ -124,11 +128,29 @@ function createModalStackNavigator<ParamList extends ParamListBase>(screens: Scr
...
+                        progress={secondOverlayProgress}
+                        positionLeftValue={modalStackOverlayWideRHPWidth}
+                    />
+                ) : null}
+                {!isSmallScreenWidth && shouldRenderSecondaryOverlay && route.name === SCREENS.RIGHT_MODAL.SEARCH_MONEY_REQUEST_REPORT && !isWideRhpFocused && !isFocused ? (
+                    <Overlay
+                        progress={secondOverlayProgress}
+                        positionLeftValue={modalStackOverlaySuperWideRHPWidth}
+                    />
+                ) : null}
```
Comment (ZhenjaHorbach) [inline src/libs/Navigation/AppNavigator/ModalStackNavigators/index.tsx]:
```
{!isSmallScreenWidth &&
    shouldRenderSecondaryOverlay &&
    route.name === SCREENS.RIGHT_MODAL.SEARCH_MONEY_REQUEST_REPORT &&
    !isFocused && (
        <Overlay
            progress={secondOverlayProgress}
            positionLeftValue={
                isWideRhpFocused
                    ? modalStackOverlayWideRHPWidth
                    : modalStackOverlaySuperWideRHPWidth
            }
        />
    )}
```

File: src/libs/Navigation/AppNavigator/Navigators/Overlay/BaseOverlay.tsx [hunk 3]
```diff
@@ -4,41 +4,32 @@ import React from 'react';
...
     /* Override the progress from useCardAnimation. Necessary for the secondary overlay */
     progress?: OverlayStylesParams;
+
+    positionLeftValue?: number | Animated.Value;
+
+    positionRightValue?: number | Animated.Value;
 };
 
-function BaseOverlay({onPress, hasMarginRight = false, progress, hasMarginLeft = false}: BaseOverlayProps) {
+function BaseOverlay({onPress, progress, positionLeftValue = -2 * variables.sideBarWidth, positionRightValue = 0}: BaseOverlayProps) {
```
Comment (ZhenjaHorbach) [inline src/libs/Navigation/AppNavigator/Navigators/Overlay/BaseOverlay.tsx:27]: Can we leave a comment on why we use these default values ?

File: src/libs/Navigation/AppNavigator/Navigators/RightModalNavigator.tsx [hunk 4]
```diff
@@ -26,12 +29,31 @@ type RightModalNavigatorProps = PlatformStackScreenProps<AuthScreensParamList, t
...
+        outputRange: [singleRHPWidth, getWideRHPWidth(windowWidth), calculateSuperWideRHPWidth(windowWidth)],
+    });
+
+    const animatedStyle = useMemo(() => {
+        return {
+            height: '100%',
+            right: 0,
+            position: 'absolute',
+            overflow: 'hidden',
+            width: shouldUseNarrowLayout ? '100%' : (animatedWidth as unknown as number),
```
Comment (ZhenjaHorbach) [inline src/libs/Navigation/AppNavigator/Navigators/RightModalNavigator.tsx]: I think it's bad practice to use unknown

File: src/components/WideRHPContextProvider/types.ts [hunk 5]
```diff
@@ -6,18 +6,30 @@ type WideRHPContextType = {
...
+    // Route keys of screens that should be displayed in super wide format
+    superWideRHPRouteKeys: string[];
+
+    // Progress of changing format: 0 - narrow, 1 - wide, 2 - super-wide
     expandedRHPProgress: Animated.Value;
 
     // Progress of the secondary overlay, the one covering wider RHP screen
     secondOverlayProgress: Animated.Value;
 
+    // Progress of the secondary overlay, the one covering wider RHP screen
```
Comment (ZhenjaHorbach) [inline src/components/WideRHPContextProvider/types.ts]: // Progress of the tertiary overlay, the one covering wider RHP screen

File: src/components/WideRHPContextProvider/types.ts [hunk 6]
```diff
@@ -6,18 +6,30 @@ type WideRHPContextType = {
...
+    thirdOverlayProgress: Animated.Value;
+
     // If the secondary overlay should be rendered. This value takes into account the delay of closing transition.
     shouldRenderSecondaryOverlay: boolean;
 
+    // If the secondary overlay should be rendered. This value takes into account the delay of closing transition.
```
Comment (ZhenjaHorbach) [inline src/components/WideRHPContextProvider/types.ts]: // If the tertiary overlay should be rendered. This value takes into account the delay of closing transition.

File: src/libs/Navigation/AppNavigator/Navigators/Overlay/BaseOverlay.tsx [hunk 7]
```diff
@@ -4,41 +4,32 @@ import React from 'react';
...
-
-    /* Whether there should be a gap on the left. Necessary for the overlay in modal stack navigator. */
-    hasMarginLeft?: boolean;
-
     /* Override the progress from useCardAnimation. Necessary for the secondary overlay */
     progress?: OverlayStylesParams;
+
+    positionLeftValue?: number | Animated.Value;
+
+    positionRightValue?: number | Animated.Value;
```
Comment (ZhenjaHorbach) [inline src/libs/Navigation/AppNavigator/Navigators/Overlay/BaseOverlay.tsx:23]: Let's add comments for new fields for consistency

File: src/libs/Navigation/AppNavigator/useRootNavigatorScreenOptions.ts [hunk 8]
```diff
@@ -28,12 +30,23 @@ const commonScreenOptions: PlatformStackNavigationOptions = {
     },
 };
 
+/* eslint-disable @typescript-eslint/no-explicit-any, @typescript-eslint/no-unsafe-assignment, @typescript-eslint/no-unsafe-argument */
+function abs(a: any) {
```
Comment (ZhenjaHorbach) [inline src/libs/Navigation/AppNavigator/useRootNavigatorScreenOptions.ts]: Can we avoid using any ?

**Comment by ZhenjaHorbach** [comment]
Small issue 
But after reloading a screen, all open modals are closed except the last one

https://github.com/user-attachments/assets/f202888d-5d21-4d93-953f-89fa4c66b96e

**Comment by ZhenjaHorbach** [comment]
I think we should improve the navigation and just close the last open modal, not all RHP modals

https://github.com/user-attachments/assets/f8a57b59-ff8e-43e4-8cb7-37a664a7dc6f

**Comment by WojtekBoman** [comment]
> After reloading a screen, all open modals are closed except the last one
> 
>  2025-11-12.10.33.13.mov

Regarding this, this case is also not supported for the Wide RHP present on staging. Rebuilding a state with multiple RHPs of different widths is a complex operation, and we must remember that we have stopped using the backTo mechanism, which would have made this much easier. If we plan to add support for rebuilding a state with multiple RHPs of different widths after a reload, let's do that as a follow-up, as this task requires more investigation. cc: @mountiny

**Comment by mountiny** [comment]
I think that is ok, refresh / reload can be considered edge case, but lets make sure they end up in OK fallback

**Review by joekaufmanexpensify — DISMISSED** [review]
Good from a product perspective 👍

**Comment by JmillsExpensify** [comment]
Took it for a quick test run and all looked good to me.

**Comment by dubielzyk-expensify** [comment]
Looking great from my end as well 👍

**Comment by ZhenjaHorbach** [comment]
The app freezes after changing the screen width with an open modal, and after pressing go back

https://github.com/user-attachments/assets/e456bbf1-fad4-4521-b79b-91e4f6a56159

**Comment by mountiny** [comment]
@JmillsExpensify @dubielzyk-expensify @dannymcclain Making a new build here, would you be able to give this one another test from product and design perspective too?

@ZhenjaHorbach are you able to test and review this one again please?

**Comment by ZhenjaHorbach** [comment]
Small bug 
And I think we can fix this later
But when I delete all expenses 
I can see Not here page for a split second

https://github.com/user-attachments/assets/3d02470e-089c-4487-a85f-436c4ab49cd3

Снимок экрана 2025-11-18 в 18 54 13

**Comment by dubielzyk-expensify** [comment]
It's looking really good to me. Couldn't break it, but curious if the others can

**Comment by ZhenjaHorbach** [comment]
And do we need to sync the selected items?

https://github.com/user-attachments/assets/ef26730c-190c-4d15-b534-c28e11f80023

**Comment by WojtekBoman** [comment]
> And do we need to sync the selected items?
> 
>  2025-11-20.10.48.29.mov

If so, I would handle it as a followup

**Comment by trjExpensify** [comment]
No, you don't need to worry about that here, as we're working on the headers/columns for the report results in another PR.

File: src/components/MoneyRequestReportView/MoneyRequestReportActionsList.tsx [hunk 9]
```diff
@@ -165,7 +166,10 @@ function MoneyRequestReportActionsList({
     const isReportArchived = useReportIsArchived(reportID);
     const canPerformWriteAction = canUserPerformWriteAction(report, isReportArchived);
 
-    const {shouldUseNarrowLayout} = useResponsiveLayout();
+    // eslint-disable-next-line rulesdir/prefer-shouldUseNarrowLayout-instead-of-isSmallScreenWidth
+    const {shouldUseNarrowLayout: shouldUseNarrowLayoutByDefault, isSmallScreenWidth} = useResponsiveLayout();
+    const {superWideRHPRouteKeys} = useContext(WideRHPContext);
+    const shouldUseNarrowLayout = shouldUseNarrowLayoutByDefault && (superWideRHPRouteKeys.length === 0 || isSmallScreenWidth);
```
Comment (mountiny) [inline src/components/MoneyRequestReportView/MoneyRequestReportActionsList.tsx:172]: I feel like this should be abstracted out

File: src/components/MoneyRequestReportView/MoneyRequestReportView.tsx [hunk 10]
```diff
@@ -211,47 +233,61 @@ function MoneyRequestReportView({report, policy, reportMetadata, shouldDisplayRe
...
-                            hasOlderActions={hasOlderActions}
-                            parentReportAction={parentReportAction}
-                            transactionThreadReportID={transactionThreadReportID}
-                        />
+                <View style={[styles.flex1, styles.flexRow]}>
+                    {shouldShowWideRHPReceipt && (
+                        <Animated.View style={styles.wideRHPMoneyRequestReceiptViewContainer}>
+                            <ScrollView contentContainerStyle={styles.wideRHPMoneyRequestReceiptViewScrollViewContainer}>
+                                <MoneyRequestReceiptView
+                                    allReports={allReports}
```
Comment (mountiny) [inline src/components/MoneyRequestReportView/MoneyRequestReportView.tsx:241]: Unrelated, but I dont understand why this component should be getting full reports collection

Comment (mountiny) [reply src/components/MoneyRequestReportView/MoneyRequestReportView.tsx:241]: the components needs parent and chat report so we could have just passed the two, I believe @adamgrzybowski is in blame so ideally SWM can take a look at this

File: src/components/Search/index.tsx [hunk 11]
```diff
@@ -737,6 +739,13 @@ function Search({
                         setOptimisticDataForTransactionThreadPreview(firstTransaction, transactionPreviewData);
                     }
                 }
+
+                if (item.transactions.length > 1) {
+                    markReportIDAsMultiTransactionExpense(reportID);
+                } else {
+                    unmarkReportIDAsMultiTransactionExpense(reportID);
```
Comment (mountiny) [inline src/components/Search/index.tsx:746]: should these be added to the useCallback deps?

File: src/components/WideRHPContextProvider/index.tsx [hunk 12]
```diff
@@ -91,77 +127,189 @@ const expenseReportSelector = (reports: OnyxCollection<Report>) => {
     );
 };
 
+function getCurrentWideRHPKeys(allWideRHPKeys: string[], lastVisibleRHPRouteKey: string | undefined) {
```
Comment (mountiny) [inline src/components/WideRHPContextProvider/index.tsx:130]: Can you add plain English explanation for this one, please?

File: src/components/WideRHPContextProvider/index.tsx [hunk 13]
```diff
@@ -91,77 +127,189 @@ const expenseReportSelector = (reports: OnyxCollection<Report>) => {
...
 
-        const lastRHPRoute = rootState.routes.find((route) => route.key === lastVisibleRHPRouteKey);
+    const isWideRHPFocused = useMemo(() => {
+        return !!focusedRouteKey && wideRHPRouteKeys.includes(focusedRouteKey);
+    }, [focusedRouteKey, wideRHPRouteKeys]);
 
-        if (!lastRHPRoute) {
-            return [];
-        }
+    const getIsSuperWideRHPBelowFocusedScreen = useCallback(
```
Comment (mountiny) [inline src/components/WideRHPContextProvider/index.tsx:205]: can you please add a comment here explaining what the value represents/ what it is for in plain english

File: src/components/WideRHPContextProvider/index.tsx [hunk 14]
```diff
@@ -91,77 +127,189 @@ const expenseReportSelector = (reports: OnyxCollection<Report>) => {
...
+            const isFocusedRouteSuperWide = isSuperWideRHPRouteName(focusedRoute.name);
 
-        const lastRHPKeys = extractNavigationKeys(lastRHPRoute.state);
-        const currentKeys = allWideRHPRouteKeys.filter((key) => lastRHPKeys.has(key));
+            return currentSuperWideRHPRouteKeys.length > 0 && !currentSuperWideRHPRouteKeys.includes(focusedRoute.key) && !isFocusedRouteSuperWide;
+        },
+        [allSuperWideRHPRouteKeys],
+    );
+
+    const getIsWideRHPBelowFocusedScreen = useCallback(
```
Comment (mountiny) [inline src/components/WideRHPContextProvider/index.tsx:226]: can you please add a comment here explaining what the value represents/ what it is for in plain english

File: src/components/WideRHPContextProvider/index.tsx [hunk 15]
```diff
@@ -176,9 +324,24 @@ function WideRHPContextProvider({children}: React.PropsWithChildren) {
...
     }, []);
 
+    const showSuperWideRHPVersion = useCallback((route: NavigationRoute) => {
+        if (!route.key) {
+            console.error(`The route passed to showWideRHPVersion should have the "key" property defined.`);
+            return;
+        }
+
+        const newKey = route.key;
+        // If the key is in the array, don't add it.
```
Comment (mountiny) [inline src/components/WideRHPContextProvider/index.tsx:339]:
```suggestion

        // If the key is in the array, don't add it.
```

**Review by mountiny — APPROVED** [review]
We should make sure we add unit tests for all the hooks and helper methods added here but since we are trying to push this one ahead for release for customers asap, lets get this to staging. Jason, @ZhenjaHorbach and design team have all tested it and generally the flows work good, no major issues found so lets see if QA can find any

**Comment by mountiny** [comment]
A _better safe than sorry_ revert PR [here](https://github.com/Expensify/App/pull/75713)

**Comment by dangrous** [comment]
Is https://github.com/Expensify/App/issues/75750 a bug or new behavior? Almost certainly comes from here. Thanks!
